Loki catalog is not correctly loaded on EnvMultinodeAllConnectors

Loki catalog is not correctly loaded on `EnvMultinodeAllConnectors`.

`EnvMultinodeAllConnectors` should load all catalogs listed [here](https://github.com/trinodb/trino/blob/ebedd6a5bd884a6574443c3388df1a3d082f079b/testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvMultinodeAllConnectors.java#L45), however there is following error for loki:
```
presto-master       | 2025-08-27T02:43:56.364+0545	INFO	cluster-ClusterId{value='68ae20062e729c9aeaefebe8', description='null'}-host1.invalid:27017	org.mongodb.driver.cluster	Exception in monitor thread while connecting to server host1.invalid:27017
presto-master       | com.mongodb.MongoSocketException: host1.invalid: Name or service not known
presto-master       | 	at com.mongodb.internal.connection.ServerAddressHelper.getSocketAddresses(ServerAddressHelper.java:75)
presto-master       | 	at com.mongodb.internal.connection.SocketStream.initializeSocket(SocketStream.java:100)
presto-master       | 	at com.mongodb.internal.connection.SocketStream.open(SocketStream.java:79)
presto-master       | 	at com.mongodb.internal.connection.InternalStreamConnection.open(InternalStreamConnection.java:233)
presto-master       | 	at com.mongodb.internal.connection.DefaultServerMonitor$ServerMonitor.lookupServerDescription(DefaultServerMonitor.java:219)
presto-master       | 	at com.mongodb.internal.connection.DefaultServerMonitor$ServerMonitor.run(DefaultServerMonitor.java:176)
presto-master       | Caused by: java.net.UnknownHostException: host1.invalid: Name or service not known
presto-master       | 	at java.base/java.net.Inet6AddressImpl.lookupAllHostAddr(Native Method)
presto-master       | 	at java.base/java.net.Inet6AddressImpl.lookupAllHostAddr(Inet6AddressImpl.java:52)
presto-master       | 	at java.base/java.net.InetAddress$PlatformResolver.lookupByName(InetAddress.java:1128)
presto-master       | 	at java.base/java.net.InetAddress.getAddressesFromNameService(InetAddress.java:1697)
presto-master       | 	at java.base/java.net.InetAddress$NameServiceAddresses.get(InetAddress.java:1056)
presto-master       | 	at java.base/java.net.InetAddress.getAllByName0(InetAddress.java:1687)
presto-master       | 	at java.base/java.net.InetAddress.getAllByName(InetAddress.java:1568)
presto-master       | 	at com.mongodb.internal.connection.DefaultInetAddressResolver.lookupByName(DefaultInetAddressResolver.java:34)
presto-master       | 	at com.mongodb.internal.connection.ServerAddressHelper.getSocketAddresses(ServerAddressHelper.java:71)
presto-master       | 	... 5 more
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -923,6 +923,7 @@ jobs:
             - suite-parquet
             - suite-oauth2
             - suite-ldap
+            - suite-loki
             - suite-compatibility
             - suite-all-connectors-smoke
             - suite-delta-lake-oss

--- a/plugin/trino-loki/pom.xml
+++ b/plugin/trino-loki/pom.xml
@@ -53,7 +53,6 @@
         <dependency>
             <groupId>io.github.jeschkies</groupId>
             <artifactId>loki-client</artifactId>
-            <version>0.0.5</version>
         </dependency>
 
         <dependency>

--- a/pom.xml
+++ b/pom.xml
@@ -789,6 +789,12 @@
             </dependency>
 
             <dependency>
+                <groupId>io.github.jeschkies</groupId>
+                <artifactId>loki-client</artifactId>
+                <version>0.0.5</version>
+            </dependency>
+
+            <dependency>
                 <groupId>io.jsonwebtoken</groupId>
                 <artifactId>jjwt-api</artifactId>
                 <version>${dep.jsonwebtoken.version}</version>

--- a/testing/trino-product-tests-groups/src/main/java/io/trino/tests/product/TestGroups.java
+++ b/testing/trino-product-tests-groups/src/main/java/io/trino/tests/product/TestGroups.java
@@ -72,6 +72,7 @@ public final class TestGroups
     public static final String LDAP_CLI = "ldap_cli";
     public static final String LDAP_AND_FILE_CLI = "ldap_and_file_cli";
     public static final String LDAP_MULTIPLE_BINDS = "ldap_multiple_binds";
+    public static final String LOKI = "loki";
     public static final String TLS = "tls";
     public static final String LARGE_QUERY = "large_query";
     public static final String KAFKA = "kafka";

--- a/testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvMultinodeLoki.java
+++ b/testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvMultinodeLoki.java
@@ -1,0 +1,67 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.tests.product.launcher.env.environment;
+
+import com.google.inject.Inject;
+import io.trino.tests.product.launcher.docker.DockerFiles;
+import io.trino.tests.product.launcher.docker.DockerFiles.ResourceProvider;
+import io.trino.tests.product.launcher.env.DockerContainer;
+import io.trino.tests.product.launcher.env.Environment.Builder;
+import io.trino.tests.product.launcher.env.EnvironmentProvider;
+import io.trino.tests.product.launcher.env.common.StandardMultinode;
+import io.trino.tests.product.launcher.env.common.TestsEnvironment;
+import io.trino.tests.product.launcher.testcontainers.PortBinder;
+import org.testcontainers.containers.wait.strategy.Wait;
+
+import java.time.Duration;
+
+import static java.util.Objects.requireNonNull;
+import static org.testcontainers.utility.MountableFile.forHostPath;
+
+@TestsEnvironment
+public class EnvMultinodeLoki
+        extends EnvironmentProvider
+{
+    public static final int LOKI_PORT = 3100;
+
+    private final ResourceProvider configDir;
+    private final PortBinder portBinder;
+
+    @Inject
+    public EnvMultinodeLoki(StandardMultinode standardMultinode, DockerFiles dockerFiles, PortBinder portBinder)
+    {
+        super(standardMultinode);
+        this.configDir = dockerFiles.getDockerFilesHostDirectory("conf/environment/multinode-loki/");
+        this.portBinder = requireNonNull(portBinder, "portBinder is null");
+    }
+
+    @Override
+    public void extendEnvironment(Builder builder)
+    {
+        builder.addConnector("loki", forHostPath(configDir.getPath("loki.properties")));
+        builder.addContainer(createLoki());
+    }
+
+    private DockerContainer createLoki()
+    {
+        DockerContainer container = new DockerContainer("grafana/loki:3.2.0", "loki")
+                .withExposedPorts(LOKI_PORT)
+                .waitingFor(Wait.forHttp("/ready").forResponsePredicate(response -> response.contains("ready")))
+                .withStartupTimeout(Duration.ofMinutes(6));
+
+        portBinder.exposePort(container, LOKI_PORT);
+
+        return container;
+    }
+}

--- a/testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/suite/suites/SuiteLoki.java
+++ b/testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/suite/suites/SuiteLoki.java
@@ -1,0 +1,37 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.tests.product.launcher.suite.suites;
+
+import com.google.common.collect.ImmutableList;
+import io.trino.tests.product.launcher.env.EnvironmentConfig;
+import io.trino.tests.product.launcher.env.environment.EnvMultinodeLoki;
+import io.trino.tests.product.launcher.suite.Suite;
+import io.trino.tests.product.launcher.suite.SuiteTestRun;
+
+import java.util.List;
+
+import static io.trino.tests.product.launcher.suite.SuiteTestRun.testOnEnvironment;
+
+public class SuiteLoki
+        extends Suite
+{
+    @Override
+    public List<SuiteTestRun> getTestRuns(EnvironmentConfig config)
+    {
+        return ImmutableList.of(
+                testOnEnvironment(EnvMultinodeLoki.class)
+                        .withGroups("configured_features", "loki")
+                        .build());
+    }
+}

--- a/testing/trino-product-tests-launcher/src/main/resources/docker/trino-product-tests/conf/environment/multinode-loki/loki.properties
+++ b/testing/trino-product-tests-launcher/src/main/resources/docker/trino-product-tests/conf/environment/multinode-loki/loki.properties
@@ -1,0 +1,2 @@
+connector.name=loki
+loki.uri=http://loki:3100

--- a/testing/trino-product-tests/pom.xml
+++ b/testing/trino-product-tests/pom.xml
@@ -99,6 +99,11 @@
         </dependency>
 
         <dependency>
+            <groupId>io.github.jeschkies</groupId>
+            <artifactId>loki-client</artifactId>
+        </dependency>
+
+        <dependency>
             <groupId>io.minio</groupId>
             <artifactId>minio</artifactId>
         </dependency>

--- a/testing/trino-product-tests/src/main/java/io/trino/tests/product/loki/TestLoki.java
+++ b/testing/trino-product-tests/src/main/java/io/trino/tests/product/loki/TestLoki.java
@@ -1,0 +1,58 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.tests.product.loki;
+
+import com.google.common.collect.ImmutableMap;
+import io.github.jeschkies.loki.client.LokiClient;
+import io.github.jeschkies.loki.client.LokiClientConfig;
+import org.testng.annotations.Test;
+
+import java.net.URI;
+import java.time.Duration;
+import java.time.Instant;
+import java.time.format.DateTimeFormatter;
+
+import static io.trino.tempto.assertions.QueryAssert.Row.row;
+import static io.trino.tests.product.TestGroups.LOKI;
+import static io.trino.tests.product.TestGroups.PROFILE_SPECIFIC_TESTS;
+import static io.trino.tests.product.utils.QueryExecutors.onTrino;
+import static java.time.ZoneOffset.UTC;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestLoki
+{
+    private static final DateTimeFormatter TIMESTAMP_FORMATTER = DateTimeFormatter.ofPattern("uuuu-MM-dd HH:mm:ss.SSS'Z'").withZone(UTC);
+
+    @Test(groups = {LOKI, PROFILE_SPECIFIC_TESTS})
+    public void testQueryRange()
+            throws Exception
+    {
+        LokiClient client = new LokiClient(new LokiClientConfig(URI.create("http://loki:3100"), Duration.ofSeconds(10)));
+
+        Instant start = Instant.now().minus(Duration.ofHours(3));
+        Instant end = start.plus(Duration.ofHours(2));
+
+        client.pushLogLine("line 1", end.minus(Duration.ofMinutes(10)), ImmutableMap.of("test", "logs_query"));
+        client.pushLogLine("line 2", end.minus(Duration.ofMinutes(5)), ImmutableMap.of("test", "logs_query"));
+        client.pushLogLine("line 3", end.minus(Duration.ofMinutes(1)), ImmutableMap.of("test", "logs_query"));
+        client.flush();
+
+        assertThat(onTrino().executeQuery("SELECT value FROM TABLE(loki.system.query_range(" +
+                                          "'{test=\"logs_query\"}'," +
+                                          "TIMESTAMP '" + TIMESTAMP_FORMATTER.format(start) + "'," +
+                                          "TIMESTAMP '" + TIMESTAMP_FORMATTER.format(end) + "'))" +
+                                          "LIMIT 1"))
+                .containsOnly(row("line 1"));
+    }
+}

